### PR TITLE
[COR-102] Run postgres pod as 'postgres' user

### PR DESCRIPTION
**Pull request by sjmiller609**
Since the container we are using, for now, was intended to run in Docker in a single "run" command, it starts as root, adjusts the permissions on the directory, then switches into postgres user to start postgres.

When working on getting psql commands to run using 'kubectl exec', I noticed an issue where by default it will exec with the root user because of this initialization step. In this PR, I refactored the initialization into an initContainer.

Also, I adjusted the test to assert the container is Ready. I noticed that the test previously only checked the container is running, but test would pass if the Pod then crashed.

### CODE REVIEW

File: coredb-operator/src/statefulset.rs [hunk 1]
```diff
@@ -26,6 +27,61 @@ pub async fn reconcile_sts(cdb: &CoreDB, ctx: Arc<Context>) -> Result<(), Error>
...
+    }]);
+
+    // This container for running postgresql
+    let postgres_container = Container {
+        env: postgres_env.clone(),
+        security_context: Some(SecurityContext {
+            run_as_user: Some(999),
+            allow_privilege_escalation: Some(false),
+            ..SecurityContext::default()
+        }),
```
Comment (sjmiller609) [inline coredb-operator/src/statefulset.rs:49]: In the docker container we are using, postgres user is ID 999.

File: coredb-operator/src/statefulset.rs [hunk 2]
```diff
@@ -26,6 +27,61 @@ pub async fn reconcile_sts(cdb: &CoreDB, ctx: Arc<Context>) -> Result<(), Error>
...
+        name: "data".to_owned(),
+        mount_path: "/var/lib/postgresql/data".to_owned(),
+        ..VolumeMount::default()
+    }]);
+
+    // This container for running postgresql
+    let postgres_container = Container {
+        env: postgres_env.clone(),
+        security_context: Some(SecurityContext {
+            run_as_user: Some(999),
```
Comment (ianstanton) [inline coredb-operator/src/statefulset.rs:46]: Would we ever want to change this? If yes, we should add a spec value like `UID`.

Comment (sjmiller609) [reply coredb-operator/src/statefulset.rs]: Good call, added that plus a test for it

File: coredb-operator/src/statefulset.rs [hunk 3]
```diff
@@ -26,6 +27,61 @@ pub async fn reconcile_sts(cdb: &CoreDB, ctx: Arc<Context>) -> Result<(), Error>
...
+    set -e
+    source /usr/local/bin/docker-entrypoint.sh
+    set -x
+    docker_setup_env
+    docker_create_db_directories
+                        "
+            .to_string(),
+        ]),
+        ..Container::default()
+    };
```
[the remark this answers is not in the document]
Comment (sjmiller609) [reply coredb-operator/src/statefulset.rs]: Ok I think I covered this, but I want to check in on it with you later

**Pull request merged**